### PR TITLE
Jira Tickets: MSDK-415, MSDK-534, MSDK-535

### PR DESCRIPTION
**Pull request by Jacob-Scheiffler**
This PR includes changes to resolve the following Jira tickets: MSDK-415, MSDK-534, and MSDK-535. As well, I added a TPU feature to select the AES key used for cipher operations and fixed a bug I found in the MAX32665 and MAX32690 SPIXF examples.

### CODE REVIEW

File: Examples/MAX32520/SPI_MasterSlave/main.c [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+#include "mxc_pins.h"
+#include "nvic_table.h"
+#include "pb.h"
+#include "spi.h"
+
+/***** Definitions *****/
+#define DATA_LEN 1024 // Words
+#define DATA_SIZE 8
+#define VALUE 0xFF
+#define SPI_SPEED 100000 // Bit Rate
```
Comment (Jake-Carter) [inline Examples/MAX32520/SPI_MasterSlave/main.c:58]: @Jacob-Scheiffler Have you tested this at higher speeds?  

On our other loopback examples there's a threshold where the example will fail, and we've thrown in a comment so customers are aware it's a limitation of the loopback setup and not the peripheral hardware.

I suspect there is a max `SPI_SPEED` for this example as well

File: Examples/MAX32520/SPI_MasterSlave/example.cfg [hunk 2]
```diff
@@ -0,0 +1,26 @@
+[Makefile]
```
Comment (Jake-Carter) [inline Examples/MAX32520/SPI_MasterSlave/example.cfg:1]: These `example.cfg` files can be deleted actually, we don't use them anymore

**Review by Jake-Carter — CHANGES_REQUESTED** [review]
Please update all project files using the latest (`git pull`) [MSDKGen](https://github.com/Analog-Devices-MSDK/MSDKGen) :)

```shell
python msdkgen.py update-all --projects SPI_MasterSlave SPIXF_ICC  --overwrite
```

The latest files have some fixes to the Makefiles and changes to the project build rules for both VS Code and Eclipse

**Review by Jake-Carter — APPROVED** [review]
Looks good

File: Examples/MAX32665/SPIXF/ramfunc.c [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+ * Copyright (C) 2022 Maxim Integrated Products, Inc., All rights Reserved.
+ * 
+ * This software is protected by copyright laws of the United States and
+ * of foreign countries. This material may also be protected by patent laws
+ * and technology transfer regulations of the United States and of foreign
+ * countries. This software is furnished under a license agreement and/or a
+ * nondisclosure agreement and may only be used or reproduced in accordance
+ * with the terms of those agreements. Dissemination of this information to
+ * any party or parties not specified in the license agreement and/or
+ * nondisclosure agreement is expressly prohibited.
```
Comment (sihyung-maxim) [inline Examples/MAX32665/SPIXF/ramfunc.c:11]: Is this file supposed to have protected copyright?

File: Examples/MAX32690/HBMC/ramfunc.c [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
+ * Copyright (C) 2022 Maxim Integrated Products, Inc., All rights Reserved.
+ * 
+ * This software is protected by copyright laws of the United States and
+ * of foreign countries. This material may also be protected by patent laws
+ * and technology transfer regulations of the United States and of foreign
+ * countries. This software is furnished under a license agreement and/or a
+ * nondisclosure agreement and may only be used or reproduced in accordance
+ * with the terms of those agreements. Dissemination of this information to
+ * any party or parties not specified in the license agreement and/or
+ * nondisclosure agreement is expressly prohibited.
```
Comment (sihyung-maxim) [inline Examples/MAX32690/HBMC/ramfunc.c:11]: Ditto on all ramfunc.c files.

File: Examples/MAX32665/SPIXF/ramfunc.c (hunk 3 above)
Comment (Jacob-Scheiffler) [reply Examples/MAX32665/SPIXF/ramfunc.c:11]: Don't think so. I bet I copied it from ME13. Good catch.

Comment (Jacob-Scheiffler) [reply Examples/MAX32665/SPIXF/ramfunc.c]: Fixed those.

**Review by sihyung-maxim — APPROVED** [review]

**Pull request merged**